server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -192,20 +219,26 @@ public static final class FlatObjectFieldType extends StringFieldType {
 
         private KeywordFieldMapper.KeywordFieldType valueAndPathFieldType;
 
-        public FlatObjectFieldType(String name, boolean isSearchable, boolean hasDocValues, Map<String, String> meta) {
-            super(name, isSearchable, false, true, TextSearchInfo.SIMPLE_MATCH_ONLY, meta);
+        public FlatObjectFieldType(
+            String name,
+            String mappedFieldTypeName,
+            boolean isSearchable,
+            boolean hasDocValues,
+            Map<String, String> meta
```

Comment:
Now that `isSearchable` and `hasDocValues` are not hard-coded, could you please add the parsing logic for them? That way someone could add a `flat_object` field type and give it doc values or make it indexed, or both. (The default, for backward compatibility, should be indexed + doc values.)

Reply:
It will by solved by the pr: https://github.com/opensearch-project/OpenSearch/pull/13853